### PR TITLE
Create Event 500 Response Error Message

### PR DESCRIPTION
**Pull request by ZachOBrien**
image

### CODE REVIEW

File: src/views/CreateEventView.vue [hunk 1]
```diff
@@ -14,10 +17,15 @@ import api from '../api/api';
 export default {
   name: 'CreateEvent',
   components: { EventForm },
+  data: () => ({
+    response500: false,
```
Comment (jackblanc) [inline src/views/CreateEventView.vue:21]: "internalError"

may want "userError" for non-500 level errors

File: src/views/CreateEventView.vue [hunk 2]
```diff
@@ -14,10 +17,15 @@ import api from '../api/api';
...
+  }),
   methods: {
     async createEvent({ event }) {
       const response = await api.createEvent(event);
-      if (response) {
+      if (!(response.status) || response.status >= 500) {
```
Comment (jackblanc) [inline src/views/CreateEventView.vue:26]: Shouldn't this really just be if there's any error? For example, if there's a 400 bad request, this code would redirect the user anyways. Might be worth generalizing this to catch any errors.

Comment (ZachOBrien) [reply src/views/CreateEventView.vue:26]: Yeah I wasn't sure about this. I figured we were catching possible 400 responses with validation but you're right we don't wan to redirect anyway in the event something weird happens. Will change.

File: src/views/CreateEventView.vue [hunk 3]
```diff
@@ -14,11 +17,18 @@ import api from '../api/api';
...
   components: { EventForm },
+  data: () => ({
+    internalError: true,
+    internalErrorMsg: '',
+  }),
   methods: {
     async createEvent({ event }) {
       const response = await api.createEvent(event);
-      if (response) {
+      if (response.status < 400) {
```
Comment (justinkonecny) [inline src/views/CreateEventView.vue:27]: Ideally, we shouldn't expect 300 status codes here either. Maybe we should just check for a status in the 200s to indicate success?

Comment (ZachOBrien) [reply src/views/CreateEventView.vue:27]: Yeah good point. I knew 300s aren't for errors but they would still be weird. I'll just check for (200 <= response.status < 300)

File: src/views/CreateEventView.vue [hunk 4]
```diff
@@ -14,11 +17,18 @@ import api from '../api/api';
 export default {
   name: 'CreateEvent',
   components: { EventForm },
+  data: () => ({
+    internalError: true,
```
Comment (ZachOBrien) [inline src/views/CreateEventView.vue:21]: Leaving a note for myself that this should default to "false"

Comment (justinkonecny) [reply src/views/CreateEventView.vue:21]: The response statuses look good, but don't forget to address this @ZachOBrien

Comment (ZachOBrien) [reply src/views/CreateEventView.vue]: thanks, just fixed

**Pull request merged**